tests/data_context/test_data_context_variables.py

```diff
@@ -416,3 +437,47 @@ def test_data_context_variables_save_config(
                 "Authorization": f"Bearer {ge_cloud_access_token}",
             },
         )
+
+
+def test_file_data_context_variables_e2e(
+    monkeypatch, file_data_context: FileDataContext, progress_bars: ProgressBarsConfig
+) -> None:
+    """
+    What does this test do and why?
+
+    Tests the E2E workflow with a FileDataContextVariables instance.
+      1. User updates certain values and sets them as attributes.
+      2. User persists changes utilizing the save_config call defined by the Variables API.
+      3. Upon reading the result config from disk, we can confirm that changes were appropriately persisted.
+
+    It is also important to note that in the case of $VARS syntax, we NEVER want to persist the underlying
+    value in order to preserve sensitive information.
+    """
+    # Prepare updated progress bars to set and serialize to disk
+    updated_progress_bars: ProgressBarsConfig = copy.deepcopy(progress_bars)
+    updated_progress_bars.globally = False
+    updated_progress_bars.profilers = True
+
+    # Prepare updated plugins directory to set and serialize to disk (ensuring we hide the true value behind $VARS syntax)
+    env_var_name: str = "MY_PLUGINS_DIRECTORY"
+    value_associated_with_env_var: str = "foo/bar/baz"
+    monkeypatch.setenv(env_var_name, value_associated_with_env_var)
+
+    # Set attributes defined above
+    file_data_context.variables.progress_bars = updated_progress_bars
+    file_data_context.variables.plugins_directory = f"${env_var_name}"
+    file_data_context.variables.save_config()
+
+    # Review great_expectations.yml where values were written and confirm changes
+    config_filepath: str = os.path.join(
+        file_data_context.root_directory, file_data_context.GE_YML
+    )
```

Comment:
Similar to above comment. Would be nice if  `file_data_context.root_directory` was already a `pathlib.Path` object.

```suggestion
    config_filepath: str = pathlib.Path(file_data_context.root_directory) / file_data_context.GE_YML
```

Or...

```python
config_filepath: str = pathlib.Path(file_data_context.root_directory).joinpath(file_data_context.GE_YML)
```